# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/ImageEncoder.h
# Commit: 204b899f436d
# Full Hash: 204b899f436d99f0eae759af3ccfcbaabfbbb428
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 09:34:54
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 3f5baab34b1f -r 204b899f436d dom/base/ImageEncoder.h
--- a/dom/base/ImageEncoder.h	Thu Jun 11 06:36:55 2020 +0000
+++ b/dom/base/ImageEncoder.h	Thu Jun 11 06:37:35 2020 +0000
@@ -20,7 +20,7 @@
 namespace mozilla {
 
 namespace layers {
-class AsyncCanvasRenderer;
+class CanvasRenderer;
 class Image;
 }  // namespace layers
 
@@ -40,7 +40,7 @@
   static nsresult ExtractData(nsAString& aType, const nsAString& aOptions,
                               const nsIntSize aSize, bool aUsePlaceholder,
                               nsICanvasRenderingContextInternal* aContext,
-                              layers::AsyncCanvasRenderer* aRenderer,
+                              layers::CanvasRenderer* aRenderer,
                               nsIInputStream** aStream);
 
   // Extracts data asynchronously. aType may change to "image/png" if we had to
@@ -86,7 +86,7 @@
       const nsAString& aType, const nsAString& aOptions, uint8_t* aImageBuffer,
       int32_t aFormat, const nsIntSize aSize, bool aUsePlaceholder,
       layers::Image* aImage, nsICanvasRenderingContextInternal* aContext,
-      layers::AsyncCanvasRenderer* aRenderer, nsIInputStream** aStream,
+      layers::CanvasRenderer* aRenderer, nsIInputStream** aStream,
       imgIEncoder* aEncoder);
 
   // Creates and returns an encoder instance of the type specified in aType.